prow/plugins/branchcleaner/branchcleaner_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package branchcleaner
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/github/fakegithub"
+)
+
+func TestBranchCleaner(t *testing.T) {
+	testcases := []struct {
+		name                 string
+		prAction             github.PullRequestEventAction
+		merged               bool
+		baseRepoFullName     string
+		headRepoFullName     string
+		repo                 string
+		org                  string
+		branchDeleteExpected bool
+	}{
+		{
+			name:                 "Opened PR nothing to do",
+			prAction:             github.PullRequestActionOpened,
+			merged:               false,
+			branchDeleteExpected: false,
+		},
+		{
+			name:                 "Closed PR unmerged nothing to do",
+			prAction:             github.PullRequestActionClosed,
+			merged:               false,
+			branchDeleteExpected: false,
+		},
+		{
+			name:                 "PR from different repo nothing to do",
+			prAction:             github.PullRequestActionClosed,
+			merged:               true,
+			baseRepoFullName:     "my-org/repo",
+			headRepoFullName:     "different-org/repo",
+			branchDeleteExpected: false,
+		},
+		{
+			name:                 "PR from same repo delete head ref",
+			prAction:             github.PullRequestActionClosed,
+			merged:               true,
+			baseRepoFullName:     "my-org/repo",
+			headRepoFullName:     "my-org/repo",
+			org:                  "my-org",
+			repo:                 "repo",
```

Comment:
Do we need the `org` and `repo` options in the test case or can we hardcode/infer it from `baseRepoFullName` below? Are there cases when these 2 values won't match that in `baseRepoFullName`?

Reply:
All the info about `BaseOrg` is indeed basically static, I've removed it from the test args